app/controllers/provider/admin/messages/inbox_controller.rb

```diff
@@ -22,11 +22,16 @@ def reply
     reply = @message.reply
     reply.attributes = message_params
 
-    reply.save!
-    reply.deliver!
-
-    flash[:notice] = 'Reply was sent.'
-    redirect_to action: :index
+    if reply.valid?
+      reply.save!
+      reply.deliver!
```

Comment:
We're calling `save!` and `deliver!` with a bang ("!") that would raise errors in case of failure and I couldn't find any rescues for those cases.
Maybe it goes like:
- We'll not see a `ActiveRecord::RecordInvalid` here because, for a reply to be created, the message should be stored on the DB. Said message should be valid (unless we have invalid messages there, of course);
- We'll not see a `StateMachines::InvalidTransition` because the just created reply will have the `unsent` initial state;

That to say that those bangs could be removed and we could do something like

```ruby
if reply.save && reply.delive
  ...
else
  ...
end

Reply:
Notice that this could only be done if we add the body validation to the Message model.

Reply:
Yes I think you are correct we already checked if reply is valid or not then only we are saving and delivering it.
i will remove the bangs we don't even need any if else conditions as well.

Reply:
we don't  need any kind of validation on body which is already cleared with the team on the JIRA itself

Reply:
Done . Removed Bangs